packages/tx/src/baseTransaction.ts

```diff
@@ -0,0 +1,195 @@
+  public readonly common: Common
+
+  constructor(txData: BaseTransactionData, txOptions: BaseTxOptions = {}) {
+    const { nonce, gasLimit, gasPrice, to, value, data } = txData
+
+    this.nonce = new BN(toBuffer(nonce))
+    this.gasPrice = new BN(toBuffer(gasPrice))
+    this.gasLimit = new BN(toBuffer(gasLimit))
+    this.to = to ? new Address(toBuffer(to)) : undefined
+    this.value = new BN(toBuffer(value))
+    this.data = toBuffer(data)
+
+    const validateCannotExceedMaxInteger = {
+      nonce: this.nonce,
+      gasPrice: this.gasPrice,
+      gasLimit: this.gasLimit,
+      value: this.value,
+    }
+
+    this.validateExceedsMaxInteger(validateCannotExceedMaxInteger)
+
+    this.common =
+      (txOptions.common &&
+        Object.assign(Object.create(Object.getPrototypeOf(txOptions.common)), txOptions.common)) ??
+      DEFAULT_COMMON
+  }
+
+  protected validateExceedsMaxInteger(validateCannotExceedMaxInteger: { [key: string]: BN }) {
+    for (const [key, value] of Object.entries(validateCannotExceedMaxInteger)) {
+      if (value && value.gt(MAX_INTEGER)) {
+        throw new Error(`${key} cannot exceed MAX_INTEGER, given ${value}`)
+      }
+    }
+  }
+
+  /**
+   * If the tx's `to` is to the creation address
+   */
+  toCreationAddress(): boolean {
+    return this.to === undefined || this.to.buf.length === 0
+  }
+
+  /**
+   * Computes a sha3-256 hash of the serialized unsigned tx, which is used to sign the transaction.
+   */
+  rawTxHash(): Buffer {
+    return this.getMessageToSign()
+  }
+
+  abstract getMessageToSign(): Buffer
+
+  /**
+   * Returns chain ID
+   */
+  getChainId(): number {
+    return this.common.chainId()
+  }
+
+  /**
+   * The amount of gas paid for the data in this tx
+   */
+  getDataFee(): BN {
+    const txDataZero = this.common.param('gasPrices', 'txDataZero')
+    const txDataNonZero = this.common.param('gasPrices', 'txDataNonZero')
+
+    let cost = 0
+    for (let i = 0; i < this.data.length; i++) {
+      this.data[i] === 0 ? (cost += txDataZero) : (cost += txDataNonZero)
+    }
+    return new BN(cost)
+  }
+
+  /**
+   * The minimum amount of gas the tx must have (DataFee + TxFee + Creation Fee)
+   */
+  getBaseFee(): BN {
+    const fee = this.getDataFee().addn(this.common.param('gasPrices', 'tx'))
+    if (this.common.gteHardfork('homestead') && this.toCreationAddress()) {
+      fee.iaddn(this.common.param('gasPrices', 'txCreation'))
+    }
+    return fee
+  }
+
+  /**
+   * The up front amount that an account must have for this transaction to be valid
+   */
+  getUpfrontCost(): BN {
+    return this.gasLimit.mul(this.gasPrice).add(this.value)
+  }
+
+  /**
+   * Checks if the transaction has the minimum amount of gas required
+   * (DataFee + TxFee + Creation Fee).
+   */
+  /**
+   * Checks if the transaction has the minimum amount of gas required
+   * (DataFee + TxFee + Creation Fee).
+   */
+  validate(): boolean
+  /* eslint-disable-next-line no-dupe-class-members */
```

Comment:
hm that's weird if you were getting lint errors for `no-dupe-class-members`, the current `transaction.ts` doesn't seem to have it.

Reply:
Yeah I was also surprised!